### PR TITLE
mockgcp: support for pubsublite 

### PR DESCRIPTION
**Pull request by justinsb**
Add support for pubsublite to mockgcp

Also fix a few of the samples.

### CODE REVIEW

**Comment by justinsb** [comment]
This one isn't particularly urgent .. just expanding our mocking capabilities

**Comment by justinsb** [comment]
/assign @acpana 

You did ask to see what one of these looks like :-)

File: pkg/test/resourcefixture/testdata/basic/pubsublite/v1alpha1/pubsublitesubscription/pubsublitesubscriptionbasic/create.yaml [hunk 1]
```diff
@@ -18,8 +18,10 @@ metadata:
   name: pubsublitesubscription-${uniqueId}
 spec:
   deliveryConfig:
-  - deliveryRequirement: DELIVER_AFTER_STORED
```
Comment (acpana) [inline pkg/test/resourcefixture/testdata/basic/pubsublite/v1alpha1/pubsublitesubscription/pubsublitesubscriptionbasic/create.yaml:21]: hm, did this not fail before?

**Review by acpana — COMMENTED** [review]
thanks for your patience Justin! I think this generally LGTM. Could you please rebase/ clear the merge conflicts when you get a chance 🙏🏼 ?

File: pkg/test/resourcefixture/testdata/basic/pubsublite/v1alpha1/pubsublitesubscription/pubsublitesubscriptionbasic/create.yaml (hunk 1 above)
Comment (justinsb) [reply pkg/test/resourcefixture/testdata/basic/pubsublite/v1alpha1/pubsublitesubscription/pubsublitesubscriptionbasic/create.yaml:21]: Great question ... @gemmahou is this on our e2e skiplist?

Comment (gemmahou) [reply pkg/test/resourcefixture/testdata/basic/pubsublite/v1alpha1/pubsublitesubscription/pubsublitesubscriptionbasic/create.yaml:21]: I believe so. I skipped alpha resources.

Comment (justinsb) [reply pkg/test/resourcefixture/testdata/basic/pubsublite/v1alpha1/pubsublitesubscription/pubsublitesubscriptionbasic/create.yaml:21]: Ah!  That'll do it :-)

**Pull request merged**